bugfix: prefix バリデーションが Linux で機能しない（CI テスト失敗）

## 概要

`SessionRotatingFileHandler` の `prefix` バリデーションが Linux 環境で正しく動作せず、CI テストが失敗している。

## 背景

PR #22 の Copilot レビュー指摘対応で `prefix` にパス区切り文字を含む場合の `ValueError` バリデーションを追加した。実装は `Path(prefix).name != prefix` で判定しているが、Linux では `\` がパス区切り文字として扱われないため、`back\slash-` のようなプレフィックスがバリデーションをすり抜ける。

Windows ローカルでは全テスト通過するが、Linux CI 環境で `test_prefix_with_path_separators_raises_value_error[back\slash-]` が失敗する。

### CI ログ

```
FAILED tests/test_session_rotating_file_handler.py::TestSessionRotatingFileHandler::test_prefix_with_path_separators_raises_value_error[back\slash-] - Failed: DID NOT RAISE <class 'ValueError'>
```

## 修正方針

`Path(prefix).name != prefix` ではなく、`os.sep` と `os.altsep` を使った明示的な文字チェックに変更する。`/` と `\` の両方を常にブロックすることで、プラットフォームに依存しないバリデーションにする。

## 影響範囲

- `src/py_common_lib/logging/session_rotating_file_handler.py` — バリデーションロジック修正
- `tests/test_session_rotating_file_handler.py` — テストがプラットフォーム非依存であることを確認
- `docs/specs/infrastructure/session-rotating-file-handler.md` — 必要に応じて制約の記述を更新

## 完了基準

- [x] prefix バリデーションを `os.sep`/`os.altsep` に依存しないプラットフォーム非依存の実装に変更
- [x] Linux CI でテスト全通過を確認

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -1,29 +1,24 @@
-name: CI
+# Quality Check — Caller
+#
+# PR 作成・更新時の品質チェック（テスト・lint・型チェック・markdownlint）。
+# Reusable Workflow (shared-workflows) を呼び出す。
 
-on:
-  pull_request:
-    branches: [main]
-  push:
-    branches: [main]
+name: Quality Check
 
 permissions:
   contents: read
 
-jobs:
-  lint:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: astral-sh/setup-uv@v6
-      - run: uv sync
-      - run: uv run ruff check .
-      - run: uv run ruff format --check .
-      - run: uv run mypy src
+on:
+  pull_request:
+    branches: [main]
 
-  test:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: astral-sh/setup-uv@v6
-      - run: uv sync
-      - run: uv run pytest --tb=short
+jobs:
+  quality:
+    uses: becky3/shared-workflows/.github/workflows/quality-check.yml@main
+    with:
+      python_version: "3.11"
+      install_command: "uv sync"
+      test_command: "uv run pytest --tb=short"
+      lint_command: "uv run ruff check . && uv run ruff format --check ."
+      typecheck_command: "uv run mypy src"
+      markdownlint: true

--- a/src/py_common_lib/logging/session_rotating_file_handler.py
+++ b/src/py_common_lib/logging/session_rotating_file_handler.py
@@ -36,7 +36,7 @@ class SessionRotatingFileHandler(logging.FileHandler):
         if max_bytes <= 0:
             msg = f"max_bytes must be positive, got {max_bytes}"
             raise ValueError(msg)
-        if Path(prefix).name != prefix:
+        if "/" in prefix or "\\" in prefix:
             msg = f"prefix must not contain path separators, got {prefix!r}"
             raise ValueError(msg)
         self._log_dir = log_dir

